protocol/mqtt_paho/v2/context.go

```diff
@@ -0,0 +1,44 @@
+package mqtt_paho
+
+import (
+	"context"
+
+	"github.com/eclipse/paho.golang/paho"
+)
+
+// ProtocolContext allows a Receiver to understand the context of a request.
+type ProtocolContext struct {
+	Topic             string
+	QoS               *byte
+	Retained          *bool
+	PacketID          *uint16
+	PublishProperties *paho.PublishProperties
+}
+
+// for more commonly used parameters, provide a constructor
```

Comment:
nit: incorrect doc string

Reply:
@embano1 How about `provide a constructor to initialize frequently used parameters?`

Reply:
Done.